src/Synapse/Synapse/Commands/DataPlaneCommands/AccessControl/NewAzureSynapseRoleAssignment.cs

```diff
@@ -93,10 +116,30 @@ public class NewAzureSynapseRoleAssignment : SynapseRoleCmdletBase
             Mandatory = true, HelpMessage = HelpMessages.PrincipalId)]
         [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = NewByWorkspaceObjectAndRoleDefinitionIdAndObjectIdParameterSet,
             Mandatory = true, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = NewByWorkspaceNameAndItemParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = NewByWorkspaceObjectAndItemParameterSet,
+            Mandatory = false, HelpMessage = HelpMessages.PrincipalId)]
         [Alias("Id", "PrincipalId")]
         [ValidateNotNullOrEmpty]
         public string ObjectId { get; set; }
 
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = NewByWorkspaceNameAndItemParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.RoleAssignmentItemType)]
+        [Parameter(ValueFromPipelineByPropertyName = false, ParameterSetName = NewByWorkspaceObjectAndItemParameterSet,
+            Mandatory = true, HelpMessage = HelpMessages.RoleAssignmentItemType)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Apache Spark pool", "Integration runtime", "Linked service", "Credential")]
```

Comment:
>Apache Spark pool", "Integration runtime", "Linked service", "Credential" [](start = 22, length = 73)

1. Nice to define a enum or const variables rather than using plain strings.
2. Do you think if we should remove the spaces?